lib/Core/Traits/PagerfantaFindServiceSearchAdaptersTrait.php

```diff
@@ -0,0 +1,114 @@
+     *
+     * @param Query $query
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createContentSearchPager(Query $query, Request $request, $maxPerPage)
+    {
+        $adapter = new ContentSearchAdapter($query, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Returns Pagerfanta pager that starts from first page
+     * and uses 'page' GET parameter from Request object
+     * configured with
+     * ContentSearchHitAdapter and FindService
+     *
+     * @param Query $query
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createContentSearchHitPager(Query $query, Request $request, $maxPerPage)
+    {
+        $adapter = new ContentSearchHitAdapter($query, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Returns Pagerfanta pager that starts from first page
+     * and uses 'page' GET parameter from Request object
+     * configured with
+     * LocationSearchAdapter and FindService
+     *
+     * @param LocationQuery $locationQuery
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createLocationSearchPager(LocationQuery $locationQuery, Request $request, $maxPerPage)
+    {
+        $adapter = new LocationSearchAdapter($locationQuery, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Returns Pagerfanta pager that starts from first page
+     * and uses 'page' GET parameter from Request object
+     * configured with
+     * LocationSearchHitAdapter and FindService
+     *
+     * @param LocationQuery $locationQuery
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function createLocationSearchHitPager(LocationQuery $locationQuery, Request $request, $maxPerPage)
+    {
+        $adapter = new LocationSearchHitAdapter($locationQuery, $this->getSite()->getFindService());
+
+        return $this->getPagerfantaPager($adapter, $request, $maxPerPage);
+    }
+
+    /**
+     * Shorthand method for creating Pagerfanta pager
+     * with preconfigured Adapter
+     *
+     * @param AdapterInterface $adapter
+     * @param Request $request
+     * @param int $maxPerPage
+     *
+     * @return Pagerfanta
+     */
+    protected function getPagerfantaPager(AdapterInterface $adapter, Request $request, $maxPerPage)
+    {
+        $pager = new Pagerfanta($adapter);
+        $pager->setNormalizeOutOfRangePages(true);
+        $pager->setMaxPerPage($maxPerPage);
+        $pager->setCurrentPage(
+            $request->query->get('page', 1)
```

Comment:
Is it a good idea to have the `Request` here? With this, I can't practically use the trait outside of controller code. It would be better to transfer the current page in method argument.